src/architecture/_GeneralModuleFiles/swig_std_array.i

```diff
@@ -0,0 +1,24 @@
+%module swig_std_array
+
+%include "std_array.i"
+
+%define STD_ARRAY(type)
+
+// Poor man's for loop
+%template() std::array<type, 1>;
+%template() std::array<type, 2>;
+%template() std::array<type, 3>;
+%template() std::array<type, 4>;
+%template() std::array<type, 5>;
+%template() std::array<type, 6>;
+%template() std::array<type, 7>;
+%template() std::array<type, 8>;
+%template() std::array<type, 9>;
+%template() std::array<type, 10>;
```

Comment:
I don't know enough swig to know if there's a better way to do this, but I think this is fine as long as it breaks in an obvious way if somebody tries to use an array with >10 elements.

Reply:
Correct, if an array larger than 10 is used, then a swig error is throwing saying the interface type is not known.

Reply:
Just committed a "smarter" but more unreadable way to do this. It also has to be expanded manually. I'd probably just stick with copy pasting `%template() std::array<type, i>`, but maybe you like the new version better.

Reply:
Howdy @juan-g-bonilla , thanks for sharing this alternate way of setting the definitions.  But, I have to agree with you, this isn't really less code, it is much harder to understand, and the user must still know where to add the dimension of the desired array.  I like the original solution being explicit and simple to read.  It doesn't require the user to immediately understand more complex use of swig coding.